espressif/vscode-esp-idf-extension
# add esp home kit sdk as framework

## Description

This PR add the `ESP-IDF: Install ESP-HomeKit-SDK` command and `idf.espHomeKitSdkPath` and `idf.espHomeKitSdkPathWin` configuration settings to save this framework.

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output

1. Click on "ESP-IDF: Install ESP-HomeKit-SDK" command
2. Select directory to clone the repository into and save it in `idf.espHomeKitSdkPath` or `idf.espHomeKitSdkPathWin` configuration setting.
3. Observe results. The repository should be cloned to the given location.
4. Try `ESP-IDF: Show Examples`, observe the ESP-HomeKit-SDK in the dropdown list, select it and see examples from this framework.
5. Create a project from this example and build it. it should built properly.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Checklist
